pkg/controller/controlplane/controlplane_controller_test.go

```diff
@@ -110,3 +129,460 @@ func TestReconcile(t *testing.T) {
+							},
+						},
+					),
+					scheme: scheme.Scheme,
+					log:    logging.New(),
+					cfnSvc: &fakeaws.MockCloudformationAPI{Status: status},
+				}
+				got, err := r.Reconcile(
+					reconcile.Request{
+						NamespacedName: types.NamespacedName{Name: "test-controlplane-name", Namespace: "test-controlplane-namespace"},
+					},
+				)
+				if (err != nil) != tt.wantErr {
+					t.Errorf("ReconcileControlPlane.Reconcile() error = %v, wantErr %v", err, tt.wantErr)
+					return
+				}
+				if !reflect.DeepEqual(got, tt.want) {
+					t.Errorf("ReconcileControlPlane.Reconcile() = %v, want %v", got, tt.want)
+				}
+			})
+		}
+	}
+}
+
+func TestReconcileControlPlane_Reconcile_CloudformationStatusComplete(t *testing.T) {
+	apis.AddToScheme(scheme.Scheme)
+	tests := []struct {
+		statuses []string
+		want     reconcile.Result
+		wantErr  bool
+	}{
+		{
+			statuses: awsHelper.CompleteStatuses,
+			want:     reconcile.Result{},
+			wantErr:  false,
+		},
+	}
+	for _, tt := range tests {
+		for _, status := range tt.statuses {
+			name := fmt.Sprintf("requeues if cloudformation status is %s", status)
+			t.Run(name, func(t *testing.T) {
+				r := &ReconcileControlPlane{
+					Client: fakeclient.NewFakeClient(
+						&clusterv1alpha1.ControlPlane{
+							ObjectMeta: metav1.ObjectMeta{
+								Name:      "test-controlplane-name",
+								Namespace: "test-controlplane-namespace",
+								Labels: map[string]string{
+									"eks.owner.name":      "test-eks-name",
+									"eks.owner.namespace": "test-eks-namespace",
+								},
+							},
+						},
+						&clusterv1alpha1.EKS{
+							ObjectMeta: metav1.ObjectMeta{
+								Name:      "test-eks-name",
+								Namespace: "test-eks-namespace",
+							},
+							Spec: clusterv1alpha1.EKSSpec{
+								ControlPlane: clusterv1alpha1.ControlPlaneSpec{
+									ClusterName: "test-clustername",
+								},
+							},
+						},
+					),
+					scheme: scheme.Scheme,
+					log:    logging.New(),
+					cfnSvc: &fakeaws.MockCloudformationAPI{Status: status},
+				}
+				got, err := r.Reconcile(
+					reconcile.Request{
+						NamespacedName: types.NamespacedName{Name: "test-controlplane-name", Namespace: "test-controlplane-namespace"},
+					},
+				)
+				if (err != nil) != tt.wantErr {
+					t.Errorf("ReconcileControlPlane.Reconcile() error = %v, wantErr %v", err, tt.wantErr)
+					return
+				}
+				if !reflect.DeepEqual(got, tt.want) {
+					t.Errorf("ReconcileControlPlane.Reconcile() = %v, want %v", got, tt.want)
+				}
+			})
+		}
+	}
+}
+
+func TestReconcileControlPlane_Reconcile_CloudformationStatusFailed(t *testing.T) {
+	apis.AddToScheme(scheme.Scheme)
+	tests := []struct {
+		statuses []string
+		want     reconcile.Result
+		wantErr  bool
+	}{
+		{
+			statuses: awsHelper.FailedStatuses,
+			want:     reconcile.Result{},
+			wantErr:  false,
+		},
+	}
+	for _, tt := range tests {
+		for _, status := range tt.statuses {
+			name := fmt.Sprintf("requeues if cloudformation status is %s", status)
```

Comment:
This says it requeues, but if err is nil, and results{} doesn't have a Requeue* then it won't.